hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockManager.java

```diff
@@ -1660,7 +1660,20 @@ private int computeReplicationWorkForBlockInternal(Block blk, int priority1)
       // abandoned block or block reopened for append
       if (bc == null || (bc.isUnderConstruction() && getBlockInfo(blk).equals(bc.getLastBlock()))) {
         // remove from neededReplications
-        neededReplications.remove(getBlockInfo(blk));
```

Comment:
can't we have a similar problem everywhere we use `getBlockInfo`? shouldn't we modify `getBlockInfo` instead?